crates/control-interface/src/types.rs

```diff
@@ -299,23 +284,25 @@ pub type RegistryCredentialMap = std::collections::HashMap<String, RegistryCrede
 /// A request to remove a link definition and detach the relevant actor
 /// from the given provider
 #[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
-pub struct RemoveLinkDefinitionRequest {
-    /// The actor's public key. This cannot be an image reference
-    #[serde(default)]
-    pub actor_id: String,
-    /// The provider contract
-    #[serde(default)]
-    pub contract_id: String,
-    /// The provider's link name
-    #[serde(default)]
-    pub link_name: String,
+pub struct RemoveInterfaceLinkDefinitionRequest {
+    /// The source component's identifier.
+    pub source_id: ComponentId,
+    /// Name of the link. Not providing this is equivalent to specifying Some("default")
+    #[serde(default = "default_link_name")]
+    pub name: LinkName,
+    /// WIT namespace of the link, e.g. `wasi` in `wasi:keyvalue/readwrite.get`
+    pub wit_namespace: WitNamespace,
+    /// WIT package of the link, e.g. `keyvalue` in `wasi:keyvalue/readwrite.get`
+    pub wit_package: WitPackage,
```

Comment:
Do we need the interface name as well? What if I want to unlink `wasi:keyvalue/readwrite` but not `wasi:keyvalue/atomic`?

Reply:
That's a good question. It's a little ambiguous and probably up to our own design, but what I was thinking was that if you put a link with interfaces then in the future you can either put the link with different interfaces, or remove it.

E.g. if you wanted to unlink `readwrite`, you'd re-put the link with just `atomic`.

Reply:
If we did want to support something like this, it would be nice to not have to specify every interface for the unlinking operation, at least at the CLI level. This command is getting pretty big:
```bash
wash link del mysource --namespace wasi --package keyvalue --interface atomic --interface eventual --name default
```

Reply:
I’m ok if we shortcut and use the package name to unlink all interfaces, but I would want a way to not unlink everything as well. I think this is good enough for now and something like this could be added later without breaking